modules/javafx.controls/src/test/java/test/com/sun/javafx/scene/control/behavior/BehaviorCleanupTest.java

```diff
@@ -49,6 +63,181 @@
+    }
+
+    /**
+     * Tests that focusOwnerListener is re-wired as expected on changing scene
+     * (here: by removing/adding the textField)
+     *
+     * Note: this tests both sceneListener and focusOwnerListener are properly updated.
+     */
+    @Test
+    public void testFocusOwnerListenerOnSceneChanged() {
+        // setup: two focusable nodes, textField focused
+        String firstWord = "some";
+        String secondWord = "text";
+        String text = firstWord + " " + secondWord;
+        TextField control = new TextField(text);
+        showControl(control, true);
+        Button button = new Button("dummy");
+        showControl(button, false);
+        control.selectNextWord();
+        assertEquals("sanity: ", secondWord, control.getSelectedText());
+        // detach textfield from scene
+        root.getChildren().remove(control);
+        assertEquals("selection unchanged after remove", secondWord, control.getSelectedText());
+        // change scene's focusOwner to another node
+        Button secondButton = new Button("another dummy");
+        showControl(secondButton, true);
+        assertEquals("selection unchanged after focusOwner change in old scene",
+                secondWord, control.getSelectedText());
+        // re-add textField
+        root.getChildren().add(control);
+        control.requestFocus();
+        assertEquals("selection changed on becoming scene's focusOwner",
+                text, control.getSelectedText());
+    }
+
+    /**
+     * Guard against regression of JDK-8116975: activate another stage must not affect
+     * selection of textField.
+     */
+    @Test
+    public void testFocusOwnerListenerSecondStage() {
+        String firstWord = "some";
+        String secondWord = "text";
+        String text = firstWord + " " + secondWord;
+        TextField control = new TextField(text);
+        showControl(control, true);
+        Button button = new Button("dummy");
+        showControl(button, false);
+        control.selectNextWord();
+        assertEquals("sanity: ", secondWord, control.getSelectedText());
+
+        // build and activate second stage
+        VBox secondRoot = new VBox(10, new Button("secondButton"));
+        Scene secondScene = new Scene(secondRoot);
+        Stage secondStage = new Stage();
+        secondStage.setScene(secondScene);
+        secondStage.show();
+        secondStage.requestFocus();
+
+        try {
+            assertTrue("sanity: ", secondStage.isFocused());
+            assertEquals("selection unchanged", secondWord, control.getSelectedText());
+            // back to first
+            stage.requestFocus();
+            assertTrue("sanity: ", stage.isFocused());
+            assertTrue("sanity: ", control.isFocused());
+            assertEquals("selection unchanged", secondWord, control.getSelectedText());
+        } finally {
+            // cleanup
+            secondStage.hide();
+        }
+
```

Comment:
minor: this empty line can be removed

Reply:
yeah, can do .. but do we have conventions about vertical whitespace? Tend to keep what suits my eyes :)